statsmodels/miscmodels/betareg.py

```diff
@@ -0,0 +1,465 @@
+            exog_precision = self.exog_precision
+
+        k_mean = self.exog.shape[1]
+        params_precision = params[k_mean:]
+        linpred = np.dot(exog_precision, params_precision)
+        phi = self.link_precision.inverse(linpred)
+
+        return phi
+
+
+    def predict_var(self, params, exog=None, exog_precision=None):
+        """predict values for conditional variance V(endog | exog)
+
+        """
+        mean = self.predict(params, exog=exog)
+        precision = self.predict_precision(params,
+                                           exog_precision=exog_precision)
+
+        var_endog = mean * (1 - mean) / (1 + precision)
+        return var_endog
+
+
+    def nloglikeobs(self, params):
+        """
+        Negative log-likelihood.
+
+        Parameters
+        ----------
+
+        params : np.ndarray
+            Parameter estimates
+        """
+        return -self._ll_br(self.endog, self.exog, self.exog_precision, params)
+
+    def _ll_br(self, y, X, Z, params):
+        nz = Z.shape[1]
+
+        Xparams = params[:-nz]
+        Zparams = params[-nz:]
+
+        mu = self.link.inverse(np.dot(X, Xparams))
+        phi = self.link_precision.inverse(np.dot(Z, Zparams))
+
+        alpha = mu * phi
+        beta = (1 - mu) * phi
+
+        if np.any(alpha <= np.finfo(float).eps): return np.array(-np.inf)
+        if np.any(beta <= np.finfo(float).eps): return np.array(-np.inf)
+
+        ll = lgamma(phi) - lgamma(mu * phi) - lgamma((1 - mu) * phi) \
+                + (mu * phi - 1) * np.log(y) + (((1 - mu) * phi) - 1) \
+                * np.log(1 - y)
+
+        return ll
+
+
+    def score(self, params):
+        """
+        Returns the score vector of the profile log-likelihood.
+
+        http://www.tandfonline.com/doi/pdf/10.1080/00949650903389993
+        """
+        sf = self.score_factor(params)
+
+        d1 = np.dot(sf[:, 0], self.exog)
+        d2 = np.dot(sf[:, 1], self.exog_precision)
+        return np.concatenate((d1, d2))
+
+
+    def score_check(self, params):
+        """inherited score with finite differences
+        """
+        return super(Beta, self).score(params)
+
+
+    def score_factor(self, params):
+        """derivative of loglikelihood function without the exog
+
+        This needs to be multiplied with the exog to obtain the score_obs
+        """
+        from scipy import special
+        digamma = special.psi
+
+        y, X, Z = self.endog, self.exog, self.exog_precision
+        nz = Z.shape[1]
+        Xparams = params[:-nz]
+        Zparams = params[-nz:]
+
+        # NO LINKS
+        mu = self.link.inverse(np.dot(X, Xparams))
+        phi = self.link_precision.inverse(np.dot(Z, Zparams))
+
+        ystar = np.log( y / (1. - y))
+        mustar = digamma(mu * phi) - digamma((1 - mu) * phi)
+        yt = np.log(1 - y)
+        mut = digamma((1 - mu) * phi) - digamma(phi)
+
+        t = 1. / self.link.deriv(mu)
+        h = 1. / self.link_precision.deriv(phi)
```

Comment:
make this less "backwards", (i.e. I prefer inverse links)
use link `derive_inverse(linpred)`